src/tabs/Tabs.tsx

```diff
@@ -135,9 +140,11 @@ export function Tabs({
   children = Children.toArray(children).filter(child => child);
   const tabs = parseTabList(children);
   // Initialize the selected tab to the first non-hidden tab
-  const [selectedIndex, setSelectedIndex] = useState<number>(
+  const [selectedStateIndex, setSelectedIndex] = useState<number>(
     tabs.findIndex(tab => !tab.hidden)
   );
```

Comment:
Can change this to initialize to the controlled state
```suggestion
  const [selectedStateIndex, setSelectedIndex] = useState<number>(
   () => {
       if (typeof controlledIndex === "number") {
          return controlledIndex;
       }
       // Select the first un-hidden index
       return tabs.findIndex(tab => !tab.hidden)
   }
  );
```